File: backend/src/main/java/com/netcracker/backend/service/UserService.java
package com.netcracker.backend.service;

import com.netcracker.backend.entity.User;

import java.util.Optional;

public interface UserService {
    User save(User user);

    Iterable<User> findAll();

    Optional<User> findById(Long id);

    Optional<User> findByUsername(String username);

    void deleteById(Long id);
}

File: frontend/src/app/modules/shared/services/user.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { User } from '../models/User';

@Injectable({
  providedIn: 'root'
})
export class UserService {

  constructor(private http: HttpClient) {

  }

  save(user: User): Observable<User> {
    return this.http.post<User>('/api/user', user);
  }

  findAll(): Observable<User[]> {
    return this.http.get<User[]>('/api/user');
  }

  findById(id: number): Observable<User> {
    return this.http.get<User>('/api/user/' + id);
  }

  findByUsername(username: string): Observable<User> {
    return this.http.get<User>('/api/user/login/' + username);
  }

  deleteById(id: number): Observable<void> {
    return this.http.delete<void>('/api/user/' + id);
  }
}
